--- .originals/problem4.c	2024-02-19 01:05:08.303985000 +0000
+++ problem4.c	2024-03-03 22:41:52.976016300 +0000
@@ -33,6 +33,11 @@
 
   result = value * factor;
 
+  if (result < 0) {
+    printf("Error: Result is negative!\n");
+    return -1;
+  }
+
   if (result == reset_key) {
     // reset_key is only usable by admins, normal users shouldn't worry about it...
     // Plus it's not like you can multiple 2 positive numbers to get a negative number anyway...
